pkgs/development/python-modules/liquidctl/default.nix

```
@@ -2,40 +2,58 @@
 , buildPythonPackage
 , fetchFromGitHub
 , pythonOlder
+, installShellFiles
 , docopt
 , hidapi
 , pyusb
 , smbus-cffi
+, i2c-tools
+, pytest
 }:
 
 buildPythonPackage rec {
   pname = "liquidctl";
-  version = "1.5.1";
+  version = "1.6.1";
   disabled = pythonOlder "3.6";
 
   src = fetchFromGitHub {
     owner  = pname;
     repo   = pname;
     rev    = "v${version}";
-    sha256 = "1l6cvm8vs2gkmg4qwg5m5vqjql1gah2vd9vs7pcj2v5hf0cm5v9x";
+    sha256 = "sha256-FYpr1mYzPc0rOE75fUNjxe/57EWl+zcbIbkqFseDhzI=";
   };
 
+  nativeBuildInputs = [ installShellFiles ];
+
   propagatedBuildInputs = [
     docopt
     hidapi
     pyusb
     smbus-cffi
+    i2c-tools
   ];
 
-  # does not contain tests
+  outputs = [ "out" "man" ];
+
+  postInstall = ''
+    installManPage liquidctl.8
+    installShellCompletion extra/completions/liquidctl.bash
+
+    mkdir -p $out/lib/udev/rules.d
+    cp extra/linux/71-liquidctl.rules $out/lib/udev/rules.d/.
+  '';
+
+  # attempts to find or create /var/run/liquidctl?
+  # AssertionError: Could not get a suitable place to store runtime data
+  #pythonImportsCheck = [ "liquidctl" ];
```

`pythonImportsCheck` still provides value, not everyone captures all their dependencies
```suggestion
  pythonImportsCheck = [ "liquidctl" ];
```

---

the comment above explains why both pythonImportCheck and doCheck are disabled
(i commented it out rather than to leave it out to make it clearer that these should get uncommented (by someone more competent than me))

---

what was the actual error? if it was `/homeless/shelter` then that just means that it want's to do IO to `$HOME`

---

the AssertionError i included in the second comment line
it originates [here](https://github.com/liquidctl/liquidctl/blob/e907649a27131b0439a84ffce7d0bef0adbb2a34/liquidctl/keyval.py#L44)

---

is this a blocking matter?

---

looks like this can be circumvented by setting `XDG_RUNTIME_DIR`
please do
```
postBuild = ''
  # needed for import
  export XDG_RUNTIME_DIR=$TMPDIR
'';
  ```

---

done, works, thanks

---

(also managed to get the included tests to work)